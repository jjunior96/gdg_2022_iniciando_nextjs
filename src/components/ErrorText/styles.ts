import styled, { css } from 'styled-components';

export const Container = styled.div``;

export const Error = styled.div`
  ${({ theme }) => css`
    height: 2rem;
    margin: 0;
    padding: 0;
    margin-top: ${theme.spacings.xxxsmall};
    margin-bottom: calc(${theme.spacings.xxxsmall} / 4); // 0.2rem

    svg {
      margin: 0;
      padding: 0;
    }

    p {
      font-size: ${theme.font.sizes.xsmall};
      color: ${theme.colors.red};
    }
  `}
`;
