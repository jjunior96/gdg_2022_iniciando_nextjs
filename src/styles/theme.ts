export default {
  grid: {
    container: '96rem',
    gutter: '3.2rem'
  },
  components: {
    sidebar: '20rem',
    header: '8rem'
  },
  border: {
    radius: '0.4rem'
  },
  box: {
    shadow: '0 8px 8px rgba(0, 0, 0, 0.2)'
  },
  font: {
    family:
      "Montserrat, -apple-system, BlinkMacSystemFont, 'Segoe UI', Roboto, Oxygen, Ubuntu, Cantarell, 'Open Sans', 'Helvetica Neue', sans-serif",
    light: 300,
    normal: 400,
    bold: 600,
    sizes: {
      xxxxsmall: '0.8rem',
      xxxsmall: '1.2rem',
      xxsmall: '1.4rem',
      xsmall: '1.6rem',
      small: '2rem',
      medium: '2.4rem',
      large: '3.2rem',
      xlarge: '4rem',
      xxlarge: '4.8rem',
      xxxlarge: '5.6rem',
      huge: '6.4rem'
    }
  },
  colors: {
    primary: '#24A66B',
    secundary: '#5855E9',
    white: '#FAFAFA',
    gray_50: '#f0efeb',
    gray_100: '#D9D9D9',
    gray_200: '#91AAB4',
    gray_300: '#3E606F',
    gray_450: '#193441',
    gray_500: '#343a40',
    yellow_300: '#F1C40F',
    purple_100: '#7F57DF',
    purple_300: '#5855E9',
    green: '#24A66B',
    red: '#E63737',
    success: '#24A66B',
    error: '#E63737',
    info: '#189CEF',
    black: '#0D0D0D'
  },
  spacings: {
    xxxsmall: '0.8rem',
    xxsmall: '1rem',
    xsmall: '1.6rem',
    small: '2.4rem',
    medium: '3.2rem',
    large: '4.0rem',
    xlarge: '4.8rem',
    xxlarge: '5.6rem'
  },
  layers: {
    base: 10,
    menu: 20,
    overlay: 30,
    modal: 40,
    alwaysOnTop: 50
  },
  modal: {
    sizes: {
      small: '40rem',
      medium: '48rem',
      large: '54rem'
    }
  },
  transition: {
    default: '0.3s ease-in-out',
    fast: '0.1s ease-in-out'
  }
} as const;
