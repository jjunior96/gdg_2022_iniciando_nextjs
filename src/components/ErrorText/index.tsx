import * as S from './styles';

export type ErrorTextProps = {
  children: React.ReactNode;
};

const ErrorText = ({ children }: ErrorTextProps) => {
  return (
    <S.Container>
      <S.Error>{children && <p>{children}</p>}</S.Error>
    </S.Container>
  );
};

export default ErrorText;
