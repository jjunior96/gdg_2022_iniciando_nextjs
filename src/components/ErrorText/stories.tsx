import { Story, Meta } from '@storybook/react/types-6-0';

import ErrorText, { ErrorTextProps } from '.';

export default {
  title: 'ErrorText',
  component: ErrorText
} as Meta;

export const Default: Story<ErrorTextProps> = (args) => (
  <div
    style={{
      width: '40rem',
      margin: '0 auto',
      paddingTop: '1rem'
    }}
  >
    <ErrorText {...args} />
  </div>
);

Default.args = {
  children: 'Text Example'
};
