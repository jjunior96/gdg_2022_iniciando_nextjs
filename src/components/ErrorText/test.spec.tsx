import { screen } from '@testing-library/react';

import { renderWithTheme } from 'utils/tests/__config__/helpers';

import theme from 'styles/theme';

import ErrorText from '.';

describe('<ErrorText />', () => {
  it('should render correctly', () => {
    const { container } = renderWithTheme(<ErrorText>Error Text</ErrorText>);

    const textElement = screen.getByText(/error text/i);

    expect(textElement).toBeInTheDocument();
    expect(textElement).toHaveStyle({
      color: theme.colors.red,
      'font-size': theme.font.sizes.xsmall
    });

    expect(container.firstChild).toMatchSnapshot();
  });
});
