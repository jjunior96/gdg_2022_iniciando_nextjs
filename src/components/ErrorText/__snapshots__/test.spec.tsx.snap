// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ErrorText /> should render correctly 1`] = `
.c0 {
  height: 2rem;
  margin: 0;
  padding: 0;
  margin-top: 0.8rem;
  margin-bottom: calc(0.8rem / 4);
}

.c0 svg {
  margin: 0;
  padding: 0;
}

.c0 p {
  font-size: 1.6rem;
  color: #E63737;
}

<div
  class=""
>
  <div
    class="c0"
  >
    <p>
      Error Text
    </p>
  </div>
</div>
`;
